### PR TITLE
Allow selecting the workspace payment method

### PR DESCRIPTION
**Pull request by meln1k**
# Description

# To-Dos

- [x] Add test coverage for new or updated functionality
- [x] Lint and test with `nox`

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://some.engineering/code-of-conduct).

### CODE REVIEW

File: fixbackend/billing_information/schemas.py [hunk 1]
```diff
@@ -40,3 +47,110 @@ def from_model(billing_entry: BillingEntry) -> "BillingEntryRead":
             period_end=billing_entry.period_end,
             nr_of_accounts_charged=billing_entry.nr_of_accounts_charged,
         )
+
+
+@deprecated("Use PaymentMethodV2 union instead")
+class PaymentMethodV1(str, Enum):
+    AWS = "aws_marketplace"
```
Comment (aquamatthias) [inline fixbackend/billing_information/schemas.py:54]: any reason we maintain this? there is no one using it..

File: fixbackend/billing_information/schemas.py [hunk 2]
```diff
@@ -40,3 +47,110 @@ def from_model(billing_entry: BillingEntry) -> "BillingEntryRead":
...
+class AwsSubscription(BaseModel):
+    method: Literal["aws_marketplace"]
+    subscription_id: SubscriptionId = Field(description="AWS Marketplace subscription identifier")
+
+
+class NoPaymentMethod(BaseModel):
+    method: Literal["none"]
+
+
+PaymentMethodV2 = Union[AwsSubscription, NoPaymentMethod]
```
Comment (aquamatthias) [inline fixbackend/billing_information/schemas.py:66]: consider to name it just PaymentMethod

File: fixbackend/billing_information/schemas.py [hunk 3]
```diff
@@ -40,3 +47,110 @@ def from_model(billing_entry: BillingEntry) -> "BillingEntryRead":
...
+
+
+class SecurityTierJson(str, Enum):
+    Free = "free"
+    Foundational = "foundational"
+    HighSecurity = "high_security"
```
Comment (aquamatthias) [inline fixbackend/billing_information/schemas.py:72]: a) SecurityTier? Isn't this more the product tier or subscription tier?
b) why is this enum called json, when it is a string?
c) I would suggest using the same literal for name and value `free = "free"`

**Review by aquamatthias — COMMENTED** [review]
LGTM - just a couple of questions and minor recommendations.

File: fixbackend/billing_information/schemas.py (hunk 2 above)
Comment (meln1k) [reply fixbackend/billing_information/schemas.py:66]: Let's not, name collisions are already bad here :)

**Review by aquamatthias — APPROVED** [review]
Thanks!

**Pull request merged**